### ISSUE TITLE
Make AsciiDoctor inception friendly (Respect the escape character \ to escape control characters)

### ISSUE THREAD

**Issue by ge0ffrey** [asciidoctor/asciidoctor#901]
**Asciidoc should use a simple and predicable escaping mechanism for control characters.**

Asciidoc has many control characters (such as *, +, `, _, ...). If any of these are proceeded by a backslash ("\"), then the character should just be rendered as is, just like the letter A would be rendered (without impacting style).

Of course, if the backslash is proceeded by a backslash, so we get a double one ("\\"), then it renders as a single backslash  ("\") and does not impact a control character that might follow it.

For example, I have this adoc text:

```
* *EntitySelector caching speedup for large datasets*: Benchmarked on MachineReassignment's B datasets.
** Tabu Search: +\+3%+ on average, +\+4%+ on largest
** Late Acceptance: +\+234%+ (+x3.34+) on average, +\+430%+ (+x5.3+) on largest
* *Generic cloning speedup for large datasets*: Benchmarked on MachineReassignment's B datasets.
** Tabu Search: +\+18%+ on average, +\+24%+ on largest
** Late Acceptance: +\+23%+ on average, +\+121%+ (+x2.21+) on largest
```

One of those lines should get rendered as:

```
Tabu Search: +3% on average, +4% on largest
```

But gets rendered as:

```
Tabu Search: \+3% on average, \+4% on largest
```

Relates to https://github.com/asciidoctor/asciidoctor/issues/759

**Comment by jirutka** [comment]
I’d like to note that this is serious problem that should not be deferred to unspecified future. Backslash escaping is basically unusable now, because it’s very unpredictable.

Let me introduce an example from the issue I’m facing right now:

```
…can be used for single level subscription and symbol `#` for multi level subscribtion.
This way you can subscribe to all temperature topics with `#/temperature`.
```

This is rendered as:

```html
…can be used for single level subscription and symbol <code><mark></mark></code> for multi level subscribtion.
This way you can subscribe to all temperature topics with <code>#/temperature</code>.
```

Okay, let’s ignore for now that the result is really surprising (I have no idea what’s going on here). Hash is a special character, so we must escape it. Let’s do it:

```
…can be used for single level subscription and symbol `\#` for multi level subscribtion.
This way you can subscribe to all temperature topics with `\#/temperature`.
```

And we get:

```html
…can be used for single level subscription and symbol <code>#</code> for multi level subscribtion.
This way you can subscribe to all temperature topics with <code>\#/temperature</code>.
```

The first `#` is now correct, but the second one is prefixed with `\`. That’s definitely not expected!

Let’s assume that author has removed the second escape, but someone else now remove the second sentence (or just make a separate paragraph or a list item from it) and (s)he get:

```html
…can be used for single level subscription and symbol <code>\#</code> for multi level subscribtion.
```

So despite (s)he didn’t touch that line at all, the rendered result is now wrong!

This leads to very frustrating user experience. In this case author used HTML entity `&num;` to avoid this problem.

> The main reason is that speed of conversion has a higher value than accuracy in the context of edge cases atm.

I bet that all writers would appreciate consistent behaviour more than saved 0.1 seconds in conversion. So the current priorities are not very reasonable…

**Comment by mojavelinux** [comment]
I agree we have to do this, at least in some predictable fashion. But it's simply not possible until we switch to a proper inline parser. I want to do it asap, but that doesn't mean it will be soon. If someone wants to begin with this work using treetop, that would certainly help speed things up.

While I certainly feel the urgency, this has been how AsciiDoc has worked for 10 years, so it's important not to view it as a regression. That certainly doesn't make it right, but it's still new territory.
